### DISCUSSION TITLE
How does ScratchAddons capture the VM?

### DISCUSSION THREAD

**Discussion by MiniCoder11** [ScratchAddons/ScratchAddons#796] [Help] [answered]
Back before ScratchAddons was started as a community project, I was working on a separate browser extension with nearly the same exact purpose but in order to capture the VM, I was using a mess of scripting that was nearly twenty lines long and not very readable.

After seeing ScratchAddons pop up after I launched my browser and attempted to check incoming messages with what I was expecting to be the old Scratch Messaging Extension, I checked and also saw that the DevTools extension had been replaced. Sure enough, I checked the popup, enabled nearly every setting just to see the new features, and witnessed my project running at 60 fps. I had made the exact same feature for my previous extension and I realized that ScratchAddons must have grabbed the VM somehow to disable compatibility mode and this GitHub page referred to the process as "traps" which made sense.

I've spent a significant amount of time looking through files and I've searched through all of the files that seem to relate to traps or the VM but couldn't find anything within those files. I'm just curious to know if there is a much more efficient way to access the VM and, if so, where can I find the file to see how it was done?

Thanks in advance!

**Comment by WorldLanguages** [comment, marked as answer]
content-scripts/prototype-handler.js

**Reply by MiniCoder11** [reply]
Thank you so much!
